zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java

```diff
@@ -133,4 +133,16 @@ public static <T> ResponseEntity<T> mapProblemToResponse(final ProblemDetail pro
         .headers(httpHeaders -> httpHeaders.setContentType(MediaType.APPLICATION_PROBLEM_JSON))
         .build();
   }
+
+  public static ResponseEntity<Object> mapUserManagementExceptionsToResponse(final Exception e) {
+    if (e instanceof IllegalArgumentException) {
+      final ProblemDetail problemDetail =
+          ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, e.getMessage());
+      return ResponseEntity.of(problemDetail).build();
+    }
+
+    final ProblemDetail problemDetail =
+        ProblemDetail.forStatusAndDetail(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    return ResponseEntity.of(problemDetail).build();
+  }
```

Comment:
🙃 Just like mentioned in https://github.com/camunda/camunda/pull/19765, we can make this shorter and consistent with the other controllers by reusing some of the methods from this utility class.